scene/gui/color_picker.cpp

```diff
@@ -2092,6 +2100,19 @@ void ColorPickerButton::pressed() {
 	}
 }
 
+void ColorPickerButton::gui_input(const Ref<InputEvent> &p_event) {
+	ERR_FAIL_COND(p_event.is_null());
+
+	Ref<InputEventMouseButton> mouse_button = p_event;
+	bool ui_accept = p_event->is_action("ui_accept", true) && !p_event->is_echo();
+	bool mouse_left_pressed = mouse_button.is_valid() && mouse_button->get_button_index() == MouseButton::LEFT && mouse_button->is_pressed();
+	if (mouse_left_pressed || ui_accept) {
+		popup_was_open = popup && popup->is_visible();
+	}
+
+	BaseButton::gui_input(p_event);
+}
```

Comment:
Why this code isn't in `pressed()` method?

Reply:
I tried that at the beginning, but I noticed that when `pressed()` is being executed, the popup is not visible anymore because you click outside of it (which hides the popup). If this code is implemented into `pressed()`, `popup->is_visible()` will always return false and that will lead to assume that the popup should be open.
To overcome that, I implemented this code into `gui_input` which gets executed before the popup is hidden due to a click was made outside of it.